pkg/http/http.go

```diff
@@ -113,28 +124,61 @@ func (s *Server) WithTracing(provider trace.TracerProvider) *Server {
 	return s
 }
 
-// WithPrometheus adds a /metrics endpoint and instrument subsequently enabled groups with general http-level metrics.
-func (s *Server) WithPrometheus() *Server {
-	// Add MW with .With instead of .Use, as .Use does not allow registering MWs after routes.
-	s.router = s.router.With(PrometheusMiddleware)
-	s.router.Handle("/metrics", promhttp.Handler())
-
+// WithInsecureTraceContext configures the server to trust parent trace IDs arriving to public-facing endpoints.
+// This will make traces generated by QuickPizza to be children of client-controlled trace IDs.
+func (s *Server) WithInsecureTraceContext() *Server {
+	s.insecureTracing = true
 	return s
 }
 
+// isPublic returns whether an endpoint should be configured public for purposes of trace linking.
+// isPublic returns false (trusted endpoint) for all endpoints if server is configured WithInsecureTraceContext.
+// Otherwise, isPublic returns false (trusted endpoint) if the X-Internal-Token header equals the secret value "secret",
+// and true (untrusted endpoint) for all remaining cases.
+func (s *Server) isPublic(r *http.Request) bool {
+	if s.insecureTracing {
+		return false
+	}
+
+	if r.Header.Get("X-Internal-Token") == "secret" {
```

Comment:
nit: Maybe we could just check if the length of this header is non-zero. Perhaps the magic "secret" constant is lil too much.